### PR TITLE
MTA-2190 Changes to header_nav to fix issue with unnecessary markup

### PR DESCRIPTION
**Pull request by gianick**
Changes to header_nav to fix issue with unnecessary markup appearing on signed-out pages. Even when there's no navTitle, navLinks or showBetaLink passed to the header_nav, the markup

```
    <div class="header-proposition">
        <div class="content">
    .
    .
    .
```

still appears on pages where the user is signed out.

### CODE REVIEW

File: src/main/twirl/uk/gov/hmrc/play/views/layouts/header_nav.scala.html [hunk 1]
```diff
@@ -18,29 +18,31 @@
...
-        @if(navLinks) {
-        <a href="#proposition-links" class="js-header-toggle menu">Menu</a>
-        }
-        <nav id="proposition-menu" class="header__menu" role="navigation">
-
-            @if(navTitleLink){
-                <a href="@navTitleLink" class="header__menu__proposition-name">@navTitle</a>
-            }else{
-                <span class="header__menu__proposition-name">@navTitle</span>
+@if(navTitle || navLinks || showBetaLink) {
```
Comment (feedmypixel) [inline src/main/twirl/uk/gov/hmrc/play/views/layouts/header_nav.scala.html]: Please use the following methods for the two options inside the if conditional:
- `navTitle.isDefined`
- `navLinks.isDefined`

**Comment by feedmypixel** [comment]
@gianick FYI when all the content has been removed from `header-proposition` there is still a  `padding-top` of `10px`. So with this change that will go which potentially may cause some undesirable movement in the header. Although that will potentially be after a page refresh/redirect (logging out for instance)
It is worth checking with your designer or @hmrc/designers that this is ok.

header-proposition

**Comment by gianick** [comment]
@feedmypixel Thanks for the FYI. I had spotted this 10px movement when removing the `header-proposition` element. The page header looks 'better' without this 10px padding since the logo appears to be vertically centered. I checked with our designer and he is fine with it.

File: src/main/twirl/uk/gov/hmrc/play/views/layouts/header_nav.scala.html (hunk 1 above)
Comment (gianick) [reply src/main/twirl/uk/gov/hmrc/play/views/layouts/header_nav.scala.html]: Updated the code. Using `isDefined` for navTitle and navLinks.

**Comment by feedmypixel** [comment]
> I checked with our designer and he is fine with it.

Has your designer raised this and had sign off from the Design community? 
FYI @hmrc/designers

**Comment by bethnoble** [comment]
> Has your designer raised this and had sign off from the Design community?

@feedmypixel I've had a look over this change with @gianick and happy for this to be made globally. It's consistent with the header on the rest of GOV.UK when there is no header-proposition.

**Comment by feedmypixel** [comment]
Thanks @bethnoble LGTM 👍

**Pull request merged**